src/main/java/smartthings/konsumer/filterchain/MessageFilterChain.java

```diff
@@ -0,0 +1,57 @@
+package smartthings.konsumer.filterchain;
+
+import kafka.message.MessageAndMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import smartthings.konsumer.MessageProcessor;
+import smartthings.konsumer.circuitbreaker.CircuitBreaker;
+
+import java.util.*;
+
+public class MessageFilterChain {
```

Comment:
Cool. No need to implement then.  I made the initial comment before reading the full PR.